dotnet-engine/UnleashEngine/UnleashEngine.cs

```diff
@@ -0,0 +1,103 @@
+﻿
+using System.Runtime.InteropServices;
+using System.Text.Json;
+
+
+namespace Unleash;
+
+public class Context
```

Comment:
Some of these properties are required in the current implementation. Do we want to make them all optional now?

Reply:
I think I have the dumb, not sure what you mean here? The implementation in .NET, Yggdrasil or our existing standard expectations of how an SDK works?

Reply:
Ah! Gotcha! That makes sense. I don't believe this is an issue for Yggdrasil. Yggdrasil doesn't really care, it'll evaluate your context whether you set these or not. Enforcing appName has more to do with how metrics are sent and how backupfiles are saved in the SDKs. So if the caller needs these it needs to set them. It's entirely reasonable for the caller to require appName to do other things but not require it for evaluating a toggle

Reply:
Right! I think this connects to my other question and shows that I didn't really understand how this to be managed. I was under the impression that these wrappers would take the place of the SDKs. In that case we're all good. Thanks for explaining 🙏🏼

Reply:
Ah, now the question makes a lot of sense.

Yep, this isn't designed to take over our SDKs. We have, for the most part, really robust and battle tested layers that do things like disk caching and http in our existing SDKs. This work is to take over the domain logic from the SDKs, not do anything else. So my intent is to eat the domain of logic of SDKs from the inside out with this - whatever the inverse of a strangler pattern is